src/falcon/falcon_c/rpo.h

```diff
@@ -0,0 +1,36 @@
+#include <stdint.h>
+#include <string.h>
+
+/* ==================================================================== */
+/*
+ * RPO hashing algorithm related structs and methods.
+ */
+
+typedef struct
+{
+    union
+    {
+        uint64_t A[12];
+        uint8_t dbuf[96];
+    } st;
+    uint64_t dptr;
+} inner_rpo128_context;
+
+void rpo128_inc_init(inner_rpo128_context *rc);
+void rpo128_inc_absorb(inner_rpo128_context *rc, const uint8_t *in, size_t len);
+void rpo128_inc_finalize(inner_rpo128_context *rc);
+void rpo128_inc_squeeze(inner_rpo128_context *rc, uint8_t *out, size_t len);
```

Comment:
Do we need `rpo128_inc_ctx_release` function? I believe SHAKE256 interface has it.

Also, would be good to add brief comments explaining the function and its parameters.

Reply:
Yes, I have added it and some comments as well.
Let me know if they are sufficient.